kernel/src/actions/visitors.rs

```diff
@@ -268,6 +270,58 @@ impl DataVisitor for RemoveVisitor {
     }
 }
 
+pub type TransactionMap = HashMap<String, Transaction>;
+
+#[derive(Default, Debug)]
+pub(crate) struct TransactionVisitor {
```

Comment:
Can we add a doc comment? In particular, explaining how the visitor handles the app id filter?